index.js

```diff
@@ -20,6 +20,10 @@ const clear = moduleId => {
 		return;
 	}
 
+	if (typeof options === 'object' && typeof options.filter === 'function' && (options.filter(filePath) === false)) {
```

Comment:
The first typeof is moot.

The filter check should also throw a useful error when it's not a boolean, as commented earlier.

Reply:
Its not moot because its javascript, options can be passed in as any type.

You would do that but it would be a performance hit to do dual typechecking.